--- /opt/rocm-5.4.2/bin/hipcc.pl.old      2023-02-03 12:45:52.705141525 +0000
+++ /opt/rocm-5.4.2/bin/hipcc.pl  2023-02-03 12:50:57.589161773 +0000
@@ -694,9 +694,9 @@
         $HIPLDFLAGS .= " -O3";
     }
     if (!$funcSupp and $optArg ne "-O0" and $hasHIP) {
-        $HIPCXXFLAGS .= " -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false";
+        $HIPCXXFLAGS .= " -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=true";
         if ($needLDFLAGS and not $needCXXFLAGS) {
-            $HIPLDFLAGS .= " -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false";
+            $HIPLDFLAGS .= " -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=true";
         }
     }
 
--- /opt/rocm-5.4.2/lib/cmake/hip/hip-config.cmake.old    2023-02-03 12:51:27.415163754 +0000
+++ /opt/rocm-5.4.2/lib/cmake/hip/hip-config.cmake        2023-02-03 12:51:38.542164492 +0000
@@ -267,7 +267,7 @@
   get_property(compilePropIsSet TARGET hip::device PROPERTY INTERFACE_COMPILE_OPTIONS SET)
 
   if (NOT compilePropIsSet AND HIP_CXX_COMPILER MATCHES ".*clang\\+\\+")
-    hip_add_interface_compile_flags(hip::device -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false)
+    hip_add_interface_compile_flags(hip::device -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=true)
   endif()
 
   if (NOT compilePropIsSet)
--- /opt/rocm-5.4.2/lib/cmake/hip-lang/hip-lang-config.cmake.old  2023-02-03 12:51:50.164165264 +0000
+++ /opt/rocm-5.4.2/lib/cmake/hip-lang/hip-lang-config.cmake      2023-02-03 12:51:55.453165616 +0000
@@ -144,7 +144,7 @@
 )
 
 set_property(TARGET hip-lang::device APPEND PROPERTY
-  INTERFACE_COMPILE_OPTIONS "$<$<COMPILE_LANGUAGE:HIP>:SHELL:-mllvm;-amdgpu-early-inline-all=true;-mllvm;-amdgpu-function-calls=false>"
+  INTERFACE_COMPILE_OPTIONS "$<$<COMPILE_LANGUAGE:HIP>:SHELL:-mllvm;-amdgpu-early-inline-all=true;-mllvm;-amdgpu-function-calls=true>"
 )
 
 if (NOT EXISTS "${AMD_DEVICE_LIBS_PREFIX}/amdgcn/bitcode")
