--- /opt/rocm/hip/bin/hipcc 2021-01-05 11:16:06.660820048 +0000
+++ /opt/rocm/hip/bin/hipcc     2021-01-05 11:16:18.363820825 +0000
@@ -818,9 +818,9 @@
         $HIPLDFLAGS .= " -O3";
     }
     if (!$funcSupp and $optArg ne "-O0" and $hasHIP) {
-        $HIPCXXFLAGS .= " -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false";
+        $HIPCXXFLAGS .= " -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=true";
         if ($needLDFLAGS and not $needCXXFLAGS) {
-            $HIPLDFLAGS .= " -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false";
+            $HIPLDFLAGS .= " -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=true";
         }
     }
 

