--- /opt/rocm/rocprim/include/rocprim/block/detail/block_scan_reduce_then_scan.hpp      2021-01-05 11:14:25.738813346 +0000
+++ /opt/rocm/rocprim/include/rocprim/block/detail/block_scan_reduce_then_scan.hpp      2021-01-05 11:14:45.432814653 +0000
@@ -507,7 +507,7 @@
             const unsigned int idx_end = idx_start + thread_reduction_size_;
 
             T thread_reduction = storage_.threads[idx_start];
-            #pragma unroll
+//            #pragma unroll
             for(unsigned int i = idx_start + 1; i < idx_end; i++)
             {
                 thread_reduction = scan_op(
@@ -527,7 +527,7 @@
             }
 
             storage_.threads[idx_start] = thread_reduction;
-            #pragma unroll
+//            #pragma unroll
             for(unsigned int i = idx_start + 1; i < idx_end; i++)
             {
                 thread_reduction = scan_op(
--- /opt/rocm/hip/bin/hipcc 2021-01-05 11:16:06.660820048 +0000
+++ /opt/rocm/hip/bin/hipcc     2021-01-05 11:16:18.363820825 +0000
@@ -818,9 +818,9 @@
         $HIPLDFLAGS .= " -O3";
     }
     if (!$funcSupp and $optArg ne "-O0" and $hasHIP) {
-        $HIPCXXFLAGS .= " -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false";
+        $HIPCXXFLAGS .= " -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=true";
         if ($needLDFLAGS and not $needCXXFLAGS) {
-            $HIPLDFLAGS .= " -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false";
+            $HIPLDFLAGS .= " -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=true";
         }
     }
 

