--- /opt/rocm-5.1.2/hip/bin/hipcc.pl.old	2022-04-26 00:25:56.000000000 +0000
+++ /opt/rocm-5.1.2/hip/bin/hipcc.pl	2022-05-17 08:15:50.338358444 +0000
@@ -687,9 +687,9 @@
         $HIPLDFLAGS .= " -O3";
     }
     if (!$funcSupp and $optArg ne "-O0" and $hasHIP) {
-        $HIPCXXFLAGS .= " -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false";
+        $HIPCXXFLAGS .= " -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=true";
         if ($needLDFLAGS and not $needCXXFLAGS) {
-            $HIPLDFLAGS .= " -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false";
+            $HIPLDFLAGS .= " -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=true";
         }
     }
 
--- /opt/rocm-5.1.2/hip/lib/cmake/hip/hip-config.cmake.old	2022-05-17 08:09:48.607334421 +0000
+++ /opt/rocm-5.1.2/hip/lib/cmake/hip/hip-config.cmake	2022-05-17 08:09:58.076335050 +0000
@@ -249,7 +249,7 @@
   get_property(compilePropIsSet TARGET hip::device PROPERTY INTERFACE_COMPILE_OPTIONS SET)
 
   if (NOT compilePropIsSet AND HIP_CXX_COMPILER MATCHES ".*clang\\+\\+")
-    hip_add_interface_compile_flags(hip::device -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false)
+    hip_add_interface_compile_flags(hip::device -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=true)
   endif()
 
   if (NOT compilePropIsSet)
--- /opt/rocm-5.1.2/hip/lib/cmake/hip-lang/hip-lang-config.cmake.old	2022-05-17 08:10:22.640336681 +0000
+++ /opt/rocm-5.1.2/hip/lib/cmake/hip-lang/hip-lang-config.cmake	2022-05-17 08:10:29.656337147 +0000
@@ -144,7 +144,7 @@
 )
 
 set_property(TARGET hip-lang::device APPEND PROPERTY
-  INTERFACE_COMPILE_OPTIONS "$<$<COMPILE_LANGUAGE:HIP>:SHELL:-mllvm;-amdgpu-early-inline-all=true;-mllvm;-amdgpu-function-calls=false>"
+  INTERFACE_COMPILE_OPTIONS "$<$<COMPILE_LANGUAGE:HIP>:SHELL:-mllvm;-amdgpu-early-inline-all=true;-mllvm;-amdgpu-+function-calls=true>"
 )
 
 if (NOT EXISTS "${AMD_DEVICE_LIBS_PREFIX}/amdgcn/bitcode")
--- /opt/rocm-5.1.2/rocprim/include/rocprim/block/detail/block_scan_reduce_then_scan.hpp.old	2022-05-17 08:17:20.770364450 +0000
+++ /opt/rocm-5.1.2/rocprim/include/rocprim/block/detail/block_scan_reduce_then_scan.hpp	2022-05-17 08:18:04.014367322 +0000
@@ -507,7 +507,7 @@
             const unsigned int idx_end = idx_start + thread_reduction_size_;
 
             T thread_reduction = storage_.threads[idx_start];
-            ROCPRIM_UNROLL
+            //ROCPRIM_UNROLL
             for(unsigned int i = idx_start + 1; i < idx_end; i++)
             {
                 thread_reduction = scan_op(
@@ -527,7 +527,7 @@
             }
 
             storage_.threads[idx_start] = thread_reduction;
-            ROCPRIM_UNROLL
+            //ROCPRIM_UNROLL
             for(unsigned int i = idx_start + 1; i < idx_end; i++)
             {
                 thread_reduction = scan_op(
